### PR TITLE
IN-1 Added initial packages

### PR DESCRIPTION
**Pull request by glachezar**

### CODE REVIEW

File: investment-app/TRINV.Api/TRINV.InvestTrackApplication/Program.cs [hunk 1]
```diff
@@ -31,6 +36,13 @@
 builder.Services.AddEndpointsApiExplorer();
 builder.Services.AddSwaggerGen();
 
+builder.Services.AddMediatR(cfg =>
```
Comment (MihaelPavlov) [inline investment-app/TRINV.Api/TRINV.InvestTrackApplication/Program.cs:39]: @glachezar  This is not the best way to initialize mediator in the program.cs. Find better way

File: investment-app/TRINV.Api/TRINV.Application/Interfaces/IInvestmentAppDbContext.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿namespace TRINV.Application.Interfaces
+{
+
+    public interface IInvestmentAppDbContext
```
Comment (MihaelPavlov) [inline investment-app/TRINV.Api/TRINV.Application/Interfaces/IInvestmentAppDbContext.cs:4]: @glachezar  Do we really need a interface for the Context? 
What are we gonna addhere ?

File: investment-app/TRINV.Api/TRINV.Infrastructure/TRINV.Infrastructure.csproj [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.25" />
```
Comment (MihaelPavlov) [inline investment-app/TRINV.Api/TRINV.Infrastructure/TRINV.Infrastructure.csproj:10]: @glachezar I think we will need and Microsoft.EntityFrameworkCore.Design

**Review by MihaelPavlov — CHANGES_REQUESTED** [review]

File: investment-app/TRINV.Api/TRINV.InvestTrackApplication/Program.cs (hunk 1 above)
Comment (glachezar) [reply investment-app/TRINV.Api/TRINV.InvestTrackApplication/Program.cs:39]: Is the example below the best way of registering of MediatR?
builder.Services.AddMediatR(cfg => cfg.RegisterServicesFromAssemblyContaining<Program>();

File: investment-app/TRINV.Api/TRINV.Application/Interfaces/IInvestmentAppDbContext.cs (hunk 2 above)
Comment (glachezar) [reply investment-app/TRINV.Api/TRINV.Application/Interfaces/IInvestmentAppDbContext.cs]: I've done some research on the matter. I have to say I was wrong. As you probably remember, I told you that the IDbContext interface is needed to retrieve data from the database. Well, that's true, but it turns out it's not the only way.

The other method involves using a specific repository pattern for each entity, with interfaces defined in the domain layer. Additionally, this approach requires using UnitOfWork, and the IUnitOfWork interface has to be defined in the application layer.

The first approach using the IDbContext interface offers faster implementation, but the downside is that if we need to switch the database type (for example, from relational to non-relational), all the handlers in the application layer would need to be rewritten.

The second approach takes longer to implement, but it means we won't be solely dependent on EfCore within the application layer. It's easier for testing and mocking. Plus, if we ever switch between relational and non-relational databases, only the repository implementations need to be rewritten.

File: investment-app/TRINV.Api/TRINV.Infrastructure/TRINV.Infrastructure.csproj (hunk 3 above)
Comment (glachezar) [reply investment-app/TRINV.Api/TRINV.Infrastructure/TRINV.Infrastructure.csproj:10]: We will need it. You are absolutely right.

File: investment-app/TRINV.Api/TRINV.InvestTrackApplication/Program.cs (hunk 1 above)
Comment (MihaelPavlov) [reply investment-app/TRINV.Api/TRINV.InvestTrackApplication/Program.cs]: Yes  !
image

**Review by MihaelPavlov — APPROVED** [review]

**Pull request merged**